nomad/node_endpoint.go

```diff
@@ -965,41 +970,51 @@ func (n *Node) DeriveVaultToken(args *structs.DeriveVaultTokenRequest,
 	//   tokens
 	snap, err := n.srv.fsm.State().Snapshot()
 	if err != nil {
-		return err
+		reply.Error = structs.NewRecoverableError(err, false)
```

Comment:
You can't serialize error types and if you return an error the reply doesn't get serialized either. So I am embedding the error in the reply